### PR TITLE
Add date picker component

### PR DESCRIPTION
**Pull request by anisha-deverashetty**
Add date picker component 

image

https://www.figma.com/design/tgFkTfmNZE0zwx44OeaohE/New-Design-System-anno-2023?node-id=16-16701&t=cX7W8hkrEqrIQiaj-0

### CODE REVIEW

File: src/Nordea/Components/DatePicker.elm [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+import Time exposing (Month, Weekday)
+
+
+type alias Config msg =
+    { hasFocus : Bool
+    , onFocus : Bool -> msg
+    , input : String
+    , onInput : String -> msg
+    , today : Date
+    , onSelect : String -> msg
```
Comment (fredrikloberg) [inline src/Nordea/Components/DatePicker.elm:71]: Could this return a Date directly?

Comment (anisha-deverashetty) [reply src/Nordea/Components/DatePicker.elm:71]: Yes, we can return date and remove the formatter.

File: src/Nordea/Components/DatePicker.elm [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+    , onMonthChange : Month -> Int -> msg
+    , selectedMonth : Maybe ( Month, Int )
```
Comment (fredrikloberg) [inline src/Nordea/Components/DatePicker.elm:73]: Could it be beneficial to hide some of these fields a little to make the component easier to use? For example create an opaque type `InternalState`:

```elm
type InternalState =
    { hasFocus : Bool
    , onFocus : Bool -> msg
    , input : String
    , onInput : String -> msg
    , today : Date
    , onMonthChange : Month -> Int -> msg
    , selectedMonth : Maybe ( Month, Int )
    }

type alias Config msg =
    { onSelect : Date -> msg
    , currentValue: Maybe Date
    , internalState : InternalState
    , onInternalStateChange : InternalState -> msg
    }

init onSelect currentValue today onSelect onInternalStateChange =
    {construct config with internal state}
``` 

This way, the user only has to be concerned with the a few fields and the component becomes free to change the internal state without needing to update any of the uses.

What do you think?

**Review by fredrikloberg — APPROVED** [review]
Great work! 🙌

File: src/Nordea/Components/DatePicker.elm (hunk 2 above)
Comment (anisha-deverashetty) [reply src/Nordea/Components/DatePicker.elm]: I have updated it based on your comment 👍

File: src/Nordea/Components/DatePicker.elm [hunk 3]
```diff
@@ -0,0 +1,410 @@
+module Nordea.Components.DatePicker exposing (InternalState(..), OptionalConfig(..), init, view)
```
Comment (fredrikloberg) [inline src/Nordea/Components/DatePicker.elm:1]: Could we make the internal state opaque? That way, if we want to make changes to it, we only need to update this file. The only way to construct it is via functions in this file, for example the init function
```suggestion
module Nordea.Components.DatePicker exposing (InternalState, OptionalConfig(..), init, view)
```

**Pull request merged**